src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -452,19 +506,31 @@ p2ptest_bank(xclDeviceHandle handle, boost::property_tree::ptree& _ptTest, std::
     return false;
   }
 
-  int counter = 0;
   XBU::ProgressBar run_test("Running Test on " + m_tag, 1024, XBU::is_escape_codes_disabled(), std::cout);
-  for(uint64_t c = 0; c < bo_size; c += chunk_size) {
-    if(!p2ptest_chunk(handle, boptr + c, addr + c, chunk_size)) {
+  if(no_dma != 0) {
+   if(!p2ptest_chunk_no_dma(handle, boptr,  mem_size,  mem_idx)){
+      _ptTest.put("status", "failed");
+      logger(_ptTest, "Error", boost::str(boost::format("P2P failed  on memory index %d")  % mem_idx));
+      free_unmap_bo(handle, boh, boptr, bo_size);
+      run_test.finish(false, "");
+      std::cout << EscapeCodes::cursor().prev_line() << EscapeCodes::cursor().clear_line();
+      return false;
+   }
```

Comment:
I don't see a counter update in this block of code. How does the progress bar update when running on a nodma platform? I think we need to revisit this piece of code.